plugin/kubernetes/controller.go

```diff
@@ -540,17 +542,18 @@ func (dns *dnsControl) GetNodeByName(ctx context.Context, name string) (*api.Nod
 
 // GetNamespaceByName returns the namespace by name. If nothing is found an error is returned.
 func (dns *dnsControl) GetNamespaceByName(name string) (*api.Namespace, error) {
-	os := dns.nsLister.List()
-	for _, o := range os {
-		ns, ok := o.(*api.Namespace)
-		if !ok {
-			continue
-		}
-		if name == ns.ObjectMeta.Name {
-			return ns, nil
-		}
+	o, exists, err := dns.nsLister.GetByKey(name)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, fmt.Errorf("namespace not found")
+	}
+	ns, ok := o.(*api.Namespace)
```

Comment:
Since the index stores `object.Namespace` ... 

```suggestion
	ns, ok := o.(*object.Namespace)
```

Function signature also needs to change to return `*object.Namespace`, and all callers of `GetNamespaceByName`, unit tests etc.

Reply:
Done. Thanks!